providers/zenvia/src/lib/zenvia.provider.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+import { ZenviaParams } from '../types/param';
+import axios from 'axios';
+
+export class ZenviaProvider implements ISmsProvider {
+  id = 'zenvia';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  private axiosInstance = axios.create();
+
+  constructor(
+    private config: {
+      apiKey: string;
+      domain: string;
+      from: string;
+    }
+  ) {}
+
+  async sendMessage(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    switch (this.config.domain) {
+      case 'SMS':
+        this.sendSMS(options);
+        break;
+      case 'WHATSAPP':
+        this.sendWhatsapp(options);
+        break;
+      default:
+        return;
+    }
+  }
+
+  private async sendWhatsapp(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const BASE_URL = 'https://api.zenvia.com/v2/channels/whatsapp/messages';
+
+    const contentParse = JSON.parse(options.content);
+
+    const data: ZenviaParams = {
+      from: options.from,
+      to: options.to,
+      contents: contentParse.contents,
+    };
+
+    const header = {
+      headers: {
+        'X-API-TOKEN': this.config.apiKey,
+        'Content-Type': 'application/json',
+      },
+    };
+
+    const url = `${BASE_URL}`;
+    const response = await this.axiosInstance.post(url, data, header);
+
+    return {
+      id: response.data.id,
+      date: new Date().toISOString(),
+    };
+  }
+
+  private async sendSMS(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const BASE_URL = 'https://api.zenvia.com/v2/channels/sms/messages';
+
+    console.log('TESTE');
+    const contentParse = JSON.parse(options.content);
+
+    console.log(contentParse);
+    const data: ZenviaParams = {
+      from: options.from,
+      to: options.to,
+      contents: contentParse.contents,
+    };
+
+    console.log(data.contents);
+
+    const header = {
+      headers: {
+        'X-API-TOKEN': this.config.apiKey,
+        'Content-Type': 'application/json',
+      },
+    };
+
+    console.log('TESTE');
+
+    const url = `${BASE_URL}`;
+    const response = await this.axiosInstance.post(url, data, header);
+
+    console.log('TESTE 2');
+
+    return {
+      id: response.data.id,
+      date: new Date().toISOString(),
+    };
```

Comment:
fazer tratamento se o retorno não for 200

Reply:
Não necessário, pois como conversamos o tratamento do novu é pelo webhook para atualizar os status.